autoPyTorch/data/tabular_feature_validator.py

```diff
@@ -161,31 +145,48 @@ def _fit(
 
```

Comment:
Can we change the name of this method to either `fit_` or `fit_validator` because it is confusing to have `_fit`, which is an implicit private method, along with `fit` in the parent class?

Reply:
I think this is a common notation used also in `sklearn`. and again we would like to stick with what it is in development

Reply:
I understood that you would like to stick to the current notation, so let's stay as it is.
But just for your information, this usage of `_fit` is not common.
More precisely, "this usage" means to define only `_fit` in a class and just inherit `fit` from a base (or parent) class. 

So I checked all the files in sklearn and the following files have `_fit` method:
```
sklearn/pipeline.py
sklearn/ensemble/_bagging.py
sklearn/neighbors/_base.py
sklearn/impute/_base.py
sklearn/cluster/_birch.py
sklearn/cluster/_bicluster.py
sklearn/linear_model/_least_angle.py
sklearn/linear_model/_stochastic_gradient.py
sklearn/feature_selection/_rfe.py
sklearn/neural_network/_rbm.py
sklearn/neural_network/_multilayer_perceptron.py
sklearn/manifold/_t_sne.py
sklearn/preprocessing/_encoders.py
sklearn/preprocessing/_data.py
sklearn/decomposition/_fastica.py
sklearn/decomposition/_pca.py
```

Except the following four classes, all of them have `fit` method in each class:
```
sklearn/cluster/_bicluster.py::SpectralCoclustering
sklearn/cluster/_bicluster.py::SpectralBiclustering
sklearn/impute/_base.py::NeighborsBase
sklearn/preprocessing/_encoders.py::_BaseEncoder
```
Two of them are base class, so we can ignore them (although it is strange to not have `def fit()` as an abstract method) and there are only two methods that only have `_fit`.
This is why it is not common to have only `_fit` in a child class.


**NOTE**
I used the following script to check all the files with the running command of `python ~/word_checker.py --path anaconda3/lib/python3.8/site-packages/sklearn --word "def _fit("`:

```python
import os
from argparse import ArgumentParser
from typing import Tuple


def _is_word_in_file(word: str, file: str) -> bool:
    with open(file, 'r') as f:
        for line in f:
            if word in line:
                return True
        else:
            return False


def _get_args() -> Tuple[str, str, str]:
    parser = ArgumentParser()
    parser.add_argument(
        '--path',
        required=True,
        type=str,
        help='The path of interest'
    )
    parser.add_argument(
        '--word',
        required=True,
        type=str,
        help='The word to check whether each file has'
    )
    parser.add_argument(
        '--ext',
        default='.py',
        type=str,
        help='The extension of files to check such as `.tex`, `.py`, `.csv`.\n'
             'You can omit the period and just parse such as `tex`, `py`, `csv` as well'
    )
    try:
        args = parser.parse_args()
    except:
        import sys
        parser.print_help()
        sys.exit()

    path, word, ext = args.path, args.word, args.ext
    if not ext.startswith('.'):
        ext = '.' + ext

    return path, word, ext


if __name__ == '__main__':
    path, word, ext = _get_args()
    path = path if 'home' in path else os.path.join(os.environ['HOME'], path)

    for root, dirs, files in os.walk(path): 
        if len(files) == 0:
            continue

        target_files = [os.path.join(root, fn) for fn in files if fn.endswith(ext)]
        passed_files = [fn for fn in target_files if _is_word_in_file(word, fn)]

        if len(passed_files) > 0:
            print('\n'.join(passed_files))
```